tests/transformer/cow_string_test.cpp

```diff
@@ -0,0 +1,80 @@
+// Unless explicitly stated otherwise all files in this repository are
+// dual-licensed under the Apache-2.0 License or BSD-3-Clause License.
+//
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
+
+#include "../test.h"
+#include <stdexcept>
+#include <transformer/common/cow_string.hpp>
+
+TEST(TestCoWString, ConstRead)
+{
+    constexpr std::string_view original = "value";
+    cow_string str(original);
+
+    EXPECT_EQ(original.length(), str.length());
+    for (size_t i = 0; i < original.length(); ++i) { EXPECT_EQ(original[i], str.at(i)); }
+
+    EXPECT_FALSE(str.modified());
+    EXPECT_NE(str.data(), nullptr);
+}
+
+TEST(TestCoWString, NonConstRead)
+{
+    constexpr std::string_view original = "value";
+    cow_string str(original);
```

Comment:
I will say that nothing is testing that a `cow_string` will assert `EXPECT_FALSE(str.modified())` when created

Reply:
At least explicitly

Reply:
Why not on this one?